### PR TITLE
Tutorial

### PR DESCRIPTION
**Pull request by fopsdev**

### CODE REVIEW

**Comment by fopsdev** [comment]
linting right now (and remving conflicts), will push again soon

**Comment by fopsdev** [comment]
hmm way too many files changed here which shouldn't.
i would have a tutorial2 branch ready which only has tutorial related stuff changed.
can we delete this pr and tutorial branch?
(i will create tutorial2 branch then and pr from there)

**Comment by Guria** [comment]
you can just force push to the same branch without need to close/reopen pr.

File: packages/cerebral-website-tutorial/package.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    "react": "^15.3.2",
+    "react-dom": "^15.3.2"
+    
+  },
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test --env=jsdom",
+    "eject": "react-scripts eject",
+    "ch01": "cd ./parts/01 && rm -rf ../../src/* ../../public/*  && cp -R * ../../ && npm start",
```
Comment (Guria) [inline packages/cerebral-website-tutorial/package.json]: `rm -rf` and `cp -R` will not work on windows. let add `rimraf` as dev dep here. also smth cp should exist as well

File: packages/cerebral-website-tutorial/parts/01/public/blaze.min.css [hunk 2]
```diff
@@ -0,0 +1 @@
...-nav__item--brand:not(:disabled):active{background-color:#1c2732;color:#fff}.c-nav__item--brand.c-nav__item--active{background-color:#2c3e50;color:#fff}.c-nav__item--info:not(:disabled):hover{background-color:#2196f3;color:#fff}.c-nav__item--info:not(:disabled):focus{box-shadow:inset 0 0 0 2px #4dabf5}.c-nav__item--info:not(:disabled):active{background-color:#0c7fda;color:#fff}.c-nav__item--info.c-nav__item--active{background-color:#2196f3;color:#fff}.c-nav__item--warning:not(:disabled):hover{background-color:#ff9800;color:#fff}.c-nav__item--warning:not(:disabled):focus{box-shadow:inset 0 0 0 2px #4dabf5}.c-nav__item--warning:not(:disabled):active{background-color:#d17d00;color:#fff}.c-nav__item--warning.c-nav__item--active{background-color:#ff9800;color:#fff}.c-nav__item--success:not(:disabled):hover{background-color:#4caf50;color:#fff}.c-nav__item--success:not(:disabled):focus{box-shadow:inset 0 0 0 2px #4dabf5}.c-nav__item--success:not(:disabled):active{background-color:#3e8f41;color:#fff}.c-nav__item--success.c-nav__item--active{background-color:#4caf50;color:#fff}.c-nav__item--error:not(:disabled):hover{background-color:#f44336;color:#fff}.c-nav__item--error:not(:disabled):foc
```
Comment (Guria) [inline packages/cerebral-website-tutorial/parts/01/public/blaze.min.css]: let's include link in html: `<link rel="stylesheet" href="https://unpkg.com/blaze">`

File: packages/cerebral-website-tutorial/parts/01/public/index.html [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="stylesheet" href="blaze.min.css">
```
Comment (Guria) [inline packages/cerebral-website-tutorial/parts/01/public/index.html]: <link rel="stylesheet" href="https://unpkg.com/blaze">

File: packages/cerebral-website-tutorial/package.json (hunk 1 above)
Comment (fopsdev) [inline packages/cerebral-website-tutorial/package.json]: i've tested it on windows and it worked

Comment (fopsdev) [inline packages/cerebral-website-tutorial/package.json]: did you see that it install shelljs?

File: packages/cerebral-website-tutorial/parts/01/public/index.html (hunk 3 above)
Comment (fopsdev) [inline packages/cerebral-website-tutorial/parts/01/public/index.html]: i had that before, but i've wanted less dependencies. thats why i've downloaded it

File: packages/cerebral-website-tutorial/parts/01/public/blaze.min.css (hunk 2 above)
Comment (fopsdev) [inline packages/cerebral-website-tutorial/parts/01/public/blaze.min.css]: i had this link before, but i've wanted less dependencies. thats why i've downloaded it and changed it everywhere.
(you never can be sure what version gets downloaded at this link: href="https://unpkg.com/blaze")

Comment (fopsdev) [inline packages/cerebral-website-tutorial/parts/01/public/blaze.min.css]: proposal:
we make just one public folder and adjust the scripts to take it always from there.
(using blaze.min.css and a static title)

Comment (Guria) [inline packages/cerebral-website-tutorial/parts/01/public/blaze.min.css]: https://unpkg.com/package@version/file

File: packages/cerebral-website-tutorial/package.json (hunk 1 above)
Comment (Guria) [inline packages/cerebral-website-tutorial/package.json]: Ah, ok, haven't note it.
For dynamic title we can replace token in html. Does shelljs have smth like sed ? :)

Comment (fopsdev) [inline packages/cerebral-website-tutorial/package.json]: yeah sed is there ( i've never used it)
https://github.com/shelljs/shelljs#sedoptions-search_regex-replacement-file--file-

Comment (fopsdev) [inline packages/cerebral-website-tutorial/package.json]: lemme see if i can get this token replacement in place :)

**Comment by fopsdev** [comment]
@guria, ok done

**Comment by Guria** [comment]
@fopsdev great. One last iteration
* remove tutorial/public from git and add it to gitignore
* use %N% as token. It shorter and uses same symbols as public url

**Comment by fopsdev** [comment]
@guria yeah makes sense. ok, done

**Pull request merged**